### ISSUE TITLE
1.8.5 fails to mount /vagrant without vagrant group

### ISSUE THREAD

**Issue by zebooka** [mitchellh/vagrant#7616]
So the issue is because BOX we use do not have "vagrant" group. Instead we have wwwrun primary group for vagrant user.

The issue is in line [14](https://github.com/mitchellh/vagrant/compare/v1.8.4...v1.8.5#diff-97d643b35101562477f4b556a379415fR14) of `plugins/guests/linux/cap/mount_virtualbox_shared_folder.rb`

`set -e` command that was added makes mount process stop at first error.
However mount process has two steps — first try using `getent group vagrant`, and then `id -g vagrant`. But the second step wont execute because of `set -e` and failing first step (due to no vagrant group).

### Vagrant version
1.8.5

### Host operating system
OS X 10.11.6

### Guest operating system
Suse 1.2.1

### Vagrantfile
Does not matter. Error is in mounting `/vagrant`

### Debug output
```
Failed to mount folders in Linux guest. This is usually because
    the "vboxsf" file system is not available. Please verify that
    the guest additions are properly installed in the guest and
    can work properly. The command attempted was:

    mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` /vagrant /vagrant
    mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` /vagrant /vagrant
```

### Expected behavior
Correctly start

### Actual behavior
Fails

### Steps to reproduce
1. Use box without vagrant group
2. Vagrant up with vagrant 1.8.5
3. No Profit.

**Comment by phyber** [comment]
In this case where you have a user `vagrant` with group `wwwrun` shouldn't you be providing at least the `group` option to `config.vm.synced_folder` or does this not address the issue?

This option is documented here: https://www.vagrantup.com/docs/synced-folders/basic_usage.html

**Comment by zebooka** [comment]
@phyber But I do not supply any options. `/vagrant` is mounted by vagrant itself automatically. I do not have any options for it. Previously it was mounted with user vagrant and his default group, but now it firstly tried to be mounted with vagrant:vagrant and then it fails and do not tries second mount way.

**Comment by codenrhoden** [comment]
I'm hitting this as well.

**Comment by jc-m** [comment]
I would like to add that this bug is also reproducible for any use of:
config.vm.synced_folder "src/", "/srv/website",
  owner: "root", group: "root"

even if the group do exist, one of the two commands will fail. In the case of ubuntu 14.4 the error is:
"gid= requires an argument (i.e. gid==<arg>)"

**Comment by jc-m** [comment]
Just so that if anyone is looking for a workaround, using the numeric user and group id avoid the problematic command. This is not going to help if the group is missing.

**Comment by tomxtobin** [comment]
FWIW, my workaround has been downgrading to 1.8.4 and patching it with [the VirtualBox 5.1 driver from 1.8.5](https://github.com/mitchellh/vagrant/commit/b57b0e0d48fe8b4196ca6b7e01bb6c1ecb4b69f9). It works fine, and should get me by until this bug gets fixed.

**Comment by benzman81** [comment]
We also face this problem. Out workaround is to hack it like this for group users:
` config.vm.synced_folder ".", "/vagrant", owner: "vagrant", group: "vagrant | getent group users | cut -d: -f3"`

**Comment by andrescevp** [comment]
same here, waiting for the fix...

**Comment by andrescevp** [comment]
by the way this fix with 1.8.4...

**Comment by Jc-L** [comment]
Replacing one line in `embedded/gems/gems/vagrant-1.8.5/plugins/guests/linux/cap/mount_virtualbox_shared_folder.rb` will silently discard the error (which was the cas previously)
-          mount_commands = ["set -e"]
+          mount_commands = [""]

Patch is attached : [vagrant-1.8.5.diff.txt](https://github.com/mitchellh/vagrant/files/411229/vagrant-1.8.5.diff.txt)

**Comment by tomascreemers** [comment]
I am also hitting this.

Note that removing `set -e` might not be sufficient: there is a bug in the second `mount` command as well. It uses the configured _group_ as argument to `id` while `id` expects a _user name_: https://github.com/mitchellh/vagrant/blob/1277f192514910f01ecd71d9acd5383867ce3372/plugins/guests/linux/cap/mount_virtualbox_shared_folder.rb#L27

Workaround: supply a group name that is also a user name. On at least OpenSuSE 12.3, "nobody" is a user name _and_ a group name. Example:
`config.vm.synced_folder ".", "/vagrant", owner: "wwwrun", group: "nobody"`

**Comment by cschug** [comment]
@tomascreemers Same observation has been made in #7662

**Comment by chrisroberts** [comment]
Fixed via #7720

**Comment by cincodenada** [comment]
For those running into this issue that can't upgrade easily: I had to re-apply this patch after a reinstall, so I slapped together a quick shell script to seek out the relevant file and apply @Jc-L's patch to it for future reference. Here's [the gist](https://gist.github.com/cincodenada/b5b1c68457f976b62c9e42b5b8e6d0ca) for anyone else who wants to use it.
